server/server.go

```diff
@@ -90,19 +92,22 @@ func (cs *cribbageServer) addReactHandlers(router *gin.Engine) {
 	router.Use(static.Serve(`/`, static.LocalFile(`./client/build`, true)))
 }
 
-func (cs *cribbageServer) Serve() {
-	router := cs.NewRouter()
-	eng, ok := router.(*gin.Engine)
-	if !ok {
-		log.Println(`router type assertion failed`)
-	}
-	cs.addWasmHandlers(eng)
-	cs.addReactHandlers(eng)
+func isLambda() bool {
+	val, ok := os.LookupEnv(`CRIBBAGE_LAMBDA`)
+	return ok && val == `true`
+}
 
-	err := eng.Run(`:` + strconv.Itoa(*restPort)) // listen and serve on 0.0.0.0:8080
-	if err != nil {
-		log.Printf("router.Run errored: %+v\n", err)
+func (cs *cribbageServer) serve() error {
+	router := gin.Default()
+	cs.addRESTRoutes(router)
+
+	if isLambda() {
```

Comment:
:mage: I would almost expect the _type_ of serving to be passed in as a config, but I don't know if I'm currently capable of succinctly expressing _why_.

Like I think it should be up to the caller to know if they should call
```go
cs.serve(config{isLambda: true})
```
or 
```go
cs.serveLambda()
```

rather than
```go
cs.serve(config{})
```
or 
```go
cs.serveStatic()
```
.

:shrug: I'm sure this is a comment I'll look back on in two years and be able to express myself better.

Reply:
I mean I think being explicit rather then having an automagic :tm: env var being checked deep in the server would be better. But if we wanted to make this a config, it seems like it should be lumped in with the other configs/inis

Reply:
Although, we can't do that because we need to know if it's a lambda before we try loading/parsing any inis :/